pkgs/development/compilers/graalvm/community-edition.nix

```diff
@@ -141,13 +178,33 @@ let
           $out/bin/java -XX:+UnlockExperimentalVMOptions -XX:+EnableJVMCI -XX:+UseJVMCICompiler HelloWorld | fgrep 'Hello World'
 
           # Ahead-Of-Time compilation
-          $out/bin/native-image --no-server HelloWorld
+          $out/bin/native-image -H:-CheckToolchain -H:+ReportExceptionStackTraces --no-server HelloWorld
           ./helloworld | fgrep 'Hello World'
 
-          # Ahead-Of-Time compilation with --static
-          $out/bin/native-image --no-server --static HelloWorld
-          ./helloworld | fgrep 'Hello World'
-        '';
+          ${lib.optionalString stdenv.isLinux ''
+            # Ahead-Of-Time compilation with --static
+            # --static flag doesn't work for darwin
+            $out/bin/native-image --no-server --static HelloWorld
+            ./helloworld | fgrep 'Hello World'
+          ''}
+
+          echo "Testing interpreted languages"
+          $out/bin/graalpython -c 'print(1 + 1)'
+          $out/bin/ruby -e 'puts(1 + 1)'
+          $out/bin/node -e 'console.log(1 + 1)'
+
+          echo '1 + 1' | $out/bin/graalpython
+
+          # TODO: `irb` on MacOS gives an error saying "Could not find OpenSSL
+          # headers, install via Homebrew or MacPorts or set OPENSSL_PREFIX", even
+          # though `openssl` is in `propagatedBuildInputs`. For more details see:
+          # https://github.com/NixOS/nixpkgs/pull/105815
+          # echo '1 + 1' | $out/bin/irb
```

Comment:
do we want to still do this check on linux?

Reply:
@taku0 pointed out in the comments above that ruby is broken on Linux as well (even though that particular test would pass).

Reply:
There maybe other ruby issues lurking on Darwin due to https://github.com/oracle/truffleruby/issues/2158.(at least on the newest version Big Sur). Looks like a fix was merged into `21.0.0`. Awesome work on this BTW!